### PR TITLE
Outline for DataPreparer object as defined pipeline object for Foresh…

### LINKED ISSUE #92: Create DataPreparer Base Class
Create DataPreparer base class and skeleton functionality for its wrapped components.

### PR DESCRIPTION
**Pull request by cchoquette**
Fixes #92 
DataPreparer base class with DataCleaner stub to demonstrate the usage of DataPreparer. This is an MVP implementation that will need to be retrofitted after more components are complete.

### CODE REVIEW

File: foreshadow/core/data_preparer.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """Predefined pipeline for the foreshadow workflow."""
+
+    def __init__(
+        self,
+        cleaner_kwargs=None,
+        intent_kwargs=None,
+        engineerer_kwargs=None,
+        preprocessor_kwargs=None,
+        reducer_kwargs=None,
+        modeler_kwargs=None,
```
Comment (adithyabsk) [inline foreshadow/core/data_preparer.py]: There wouldn't be a modeler here, it is a part of the Foreshadow object

File: foreshadow/core/data_preparer.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+from foreshadow.cleaners.data_cleaner import DataCleaner as _DataCleaner
+
+
+def _none_to_dict(**kwargs):
+    """Transform input kwarg to valid dict, handling sentinel value.
+
+    Accepts a single kwarg by default.
+
+    Args:
+        **kwargs: the kwarg value to ensure is proper format for kwargs.
```
Comment (adithyabsk) [inline foreshadow/core/data_preparer.py]: Any reason the input is **kwargs when it only takes a single input dict.

Comment (cchoquette) [reply foreshadow/core/data_preparer.py]: I fixed this in the next PR -> realistically the two should really be combined

File: pytest.ini [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [pytest]
-addopts = -v -x --pyargs foreshadow --cov=foreshadow --xdoc
-;addopts = -s -v --xdoc
+;addopts = -v -x --pyargs foreshadow --cov=foreshadow --xdoc
+addopts = -s -v --xdoc
```
[the remark this answers is not in the document]
Comment (cchoquette) [reply pytest.ini]: Fixed

File: foreshadow/core/data_preparer.py (hunk 1 above)
Comment (cchoquette) [reply foreshadow/core/data_preparer.py]: Will remove - you're right

**Pull request merged**